core/src/Cuda/Kokkos_Cuda_KernelLaunch.hpp

```diff
@@ -205,15 +207,38 @@ modify_launch_configuration_if_desired_occupancy_is_specified(
 
```

Comment:
we also should add here the same calculation as in the other one. just using the different active blocks.